sql/api/src/main/scala/org/apache/spark/sql/types/StringType.scala

```diff
@@ -65,9 +64,41 @@ class StringType private(val collationId: Int) extends AtomicType with Serializa
 }
 
 /**
+ * Use StringType for expressions supporting only binary collation.
+ *
  * @since 1.3.0
  */
 @Stable
 case object StringType extends StringType(0) {
   def apply(collationId: Int): StringType = new StringType(collationId)
 }
+
+/**
+ * Use StringTypeBinary for expressions supporting only binary collation.
+ */
+case object StringTypeBinary extends AbstractDataType {
```

Comment:
Does it make sense to move this into `CollationTypeConstraints.scala`?
I would suggest to at least be explicit and say that these are not actual types that can be constructed/used externally.

Reply:
+1